src/jemalloc.c

```diff
@@ -2709,6 +2690,9 @@ je_realloc(void *ptr, size_t arg_size) {
 			ret = iralloc(tsd, ptr, old_usize, size, 0, false,
 			    &hook_args);
 		}
+		if (unlikely(ret == NULL)) {
+			thread_event_rollback(tsd, usize);
```

Comment:
nit: merge this to the NULL check after this branch.

Reply:
That was what I had first, but then I felt that it looked so far away.

I ended up restructuring `realloc()`, so that the execution flow looks more natural and there's only one `ret == NULL` branch.